api/session/auth_session.py

```diff
@@ -158,3 +158,71 @@ def create_session_details_with_token(
         session_details.update({"token": create_token(session_details)})
         session.update(session_details)
         return session_details
+
+    @classmethod
+    def create_session_and_redirect_via_magic_link(
+        cls,
+        account: "Account",
+        redirect_url: str,
+        timeout_seconds: int = Config.FSD_SESSION_TIMEOUT_SECONDS,
+    ):
+        """
+        Sets a user session token in the client for a given account_id
+        and then redirects to a given url
+        :param account: The account object of the user we are authenticating
+        :param redirect_url: The url to redirect them to after session creation
+        :param timeout_seconds: (int, optional)
+        The session TTL to set in seconds
+        :return: 302 redirect
+        """
+        try:
+            session_details = (
+                cls.create_session_details_with_token_via_magic_link(
+                    account, timeout_seconds=timeout_seconds
+                )
+            )
+            response = make_response(redirect(redirect_url), 302)
+            expiry = datetime.now() + timedelta(seconds=timeout_seconds)
+            response.set_cookie(
+                Config.FSD_USER_TOKEN_COOKIE_NAME,
+                session_details["token"],
+                domain=Config.COOKIE_DOMAIN,
+                expires=expiry,
+                secure=Config.SESSION_COOKIE_SECURE,
+                samesite=Config.FSD_USER_TOKEN_COOKIE_SAMESITE,
+                httponly=Config.SESSION_COOKIE_HTTPONLY,
+            )
+            current_app.logger.info(
+                f"User logged in to account : {account.id}"
+            )
+            return response
+        except SessionCreateError as e:
+            error_response(404, str(e))
+
+    @classmethod
+    def create_session_details_with_token_via_magic_link(
```

Comment:
Rather than copying the whole function can we pass a parameter to it instead and vary based on the auth method?>

Reply:
passing an argument to the function now that determines whether it's being called via magic-link routes.py 👍🏾